backend/EasySpeak.Core/EasySpeak.Core.WebAPI/Middlewares/FirebaseAuthMiddleware.cs

```diff
@@ -0,0 +1,43 @@
+﻿using EasySpeak.Core.BLL.Interfaces;
+using FirebaseAdmin.Auth;
+
+namespace EasySpeak.Core.WebAPI.Middlewares
+{
+    public class FirebaseAuthMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public FirebaseAuthMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context, IFirebaseAuthService firebaseAuthService)
+        {
+            var authToken = context.Request.Headers["Authorization"].ToString();
+
+            if (!string.IsNullOrEmpty(authToken) && authToken.StartsWith("Bearer "))
+            {
+                var idToken = authToken.Substring("Bearer ".Length);
+
+                try
+                {
```

Comment:
We don't need try/catch we have Global Error Handler

Reply:
Remove this one